From 5a038c0a3563a334601e88250242584672c92bc2 Mon Sep 17 00:00:00 2001
From: snodlop <samkahn02@yahoo.com>
Date: Sat, 17 Jan 2026 18:51:41 -0800
Subject: [PATCH 15/16] buzzwole grunts back to back fight

pbbbbbbbbbbbbbbbt what the title says. this was hell to do and the cosmetic stuff still isn't done yet, but the battle mechanically works
---
 data/maps/PetalburgWoods/map.json    | 26 +++++++++++++++
 data/maps/PetalburgWoods/scripts.inc | 45 +++++++++++++++++++++++++
 include/constants/flags.h            |  6 ++--
 include/constants/opponents.h        |  6 ++--
 src/data/trainers.party              | 36 ++++++++++++++++++++
 src/data/wild_encounters.json        | 50 +++++++++++++---------------
 6 files changed, 138 insertions(+), 31 deletions(-)

diff --git a/data/maps/PetalburgWoods/map.json b/data/maps/PetalburgWoods/map.json
index 35deb2acf5..9087627f8d 100644
--- a/data/maps/PetalburgWoods/map.json
+++ b/data/maps/PetalburgWoods/map.json
@@ -184,6 +184,32 @@
       "trainer_sight_or_berry_tree_id": "0",
       "script": "PetalburgWoods_EventScript_Girl",
       "flag": "0"
+    },
+    {
+      "graphics_id": "OBJ_EVENT_GFX_AQUA_MEMBER_M",
+      "x": 47,
+      "y": 22,
+      "elevation": 0,
+      "movement_type": "MOVEMENT_TYPE_FACE_RIGHT",
+      "movement_range_x": 0,
+      "movement_range_y": 0,
+      "trainer_type": "TRAINER_TYPE_NORMAL",
+      "trainer_sight_or_berry_tree_id": "0",
+      "script": "PetalburgWoods_BackToBackTrainer_EventScript",
+      "flag": "FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_1"
+    },
+    {
+      "graphics_id": "OBJ_EVENT_GFX_AQUA_MEMBER_F",
+      "x": 47,
+      "y": 23,
+      "elevation": 0,
+      "movement_type": "MOVEMENT_TYPE_FACE_RIGHT",
+      "movement_range_x": 0,
+      "movement_range_y": 0,
+      "trainer_type": "TRAINER_TYPE_NORMAL",
+      "trainer_sight_or_berry_tree_id": "0",
+      "script": "PetalburgWoods_BackToBackTrainer_EventScript",
+      "flag": "FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_2"
     }
   ],
   "warp_events": [
diff --git a/data/maps/PetalburgWoods/scripts.inc b/data/maps/PetalburgWoods/scripts.inc
index 14cf1aabb4..97e3eb2e32 100644
--- a/data/maps/PetalburgWoods/scripts.inc
+++ b/data/maps/PetalburgWoods/scripts.inc
@@ -315,6 +315,37 @@ PetalburgWoods_EventScript_JamesRematch::
 	msgbox PetalburgWoods_Text_IveBeenTrainingHard, MSGBOX_AUTOCLOSE
 	end
 
+PetalburgWoods_BackToBackTrainer_EventScript::
+    lock
+    faceplayer
+
+    goto_if_set FLAG_BUZZWOLE_BACK_TO_BACK_TRAINERS_DEFEATED, BackToBackTrainer_Done
+
+    msgbox BackToBack_Text_Intro1, MSGBOX_DEFAULT
+
+    trainerbattle_no_intro TRAINER_BUZZWOLE_GRUNT_1, BackToBack_Text_Lose1
+
+    msgbox BackToBack_Text_Before2, MSGBOX_DEFAULT
+
+    trainerbattle_no_intro TRAINER_BUZZWOLE_GRUNT_2, BackToBack_Text_Lose2
+
+    setflag FLAG_BUZZWOLE_BACK_TO_BACK_TRAINERS_DEFEATED
+
+	msgbox BackToBack_Text_After, MSGBOX_DEFAULT
+
+	setflag FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_1
+    setflag FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_2
+    removeobject 14
+    removeobject 15
+
+    release
+    end
+
+BackToBackTrainer_Done:
+    release
+    end
+
+
 PetalburgWoods_Text_NotAOneToBeFound:
 	.string "Hmmm…\n"
 	.string "Not a one to be found…$"
@@ -471,3 +502,17 @@ PetalburgWoods_Text_TrainerTipsPP:
 	.string "If a POKéMON runs out of PP, it must be\n"
 	.string "taken to a POKéMON CENTER.$"
 
+BackToBack_Text_Intro1:
+    .string "Dialogue Before back-to-back!$"
+
+BackToBack_Text_Before2:
+    .string "Dialogue in between fights!$"
+
+BackToBack_Text_Lose1:
+	.string "I lost!$"
+
+BackToBack_Text_Lose2:
+	.string "I also lost!$"
+
+BackToBack_Text_After:
+    .string "Dialogue after$"
diff --git a/include/constants/flags.h b/include/constants/flags.h
index 24edb07539..7a68a1beaf 100644
--- a/include/constants/flags.h
+++ b/include/constants/flags.h
@@ -1222,9 +1222,9 @@
 #define FLAG_ITEM_SAFARI_ZONE_NORTH_EAST_NUGGET                     0x491
 #define FLAG_ITEM_SAFARI_ZONE_SOUTH_EAST_BIG_PEARL                  0x492
 
-#define FLAG_UNUSED_0x493                                           0x493 // Unused Flag
-#define FLAG_UNUSED_0x494                                           0x494 // Unused Flag
-#define FLAG_UNUSED_0x495                                           0x495 // Unused Flag
+#define FLAG_BUZZWOLE_BACK_TO_BACK_TRAINERS_DEFEATED                0x493 // Unused Flag
+#define FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_1                   0x494 // Unused Flag
+#define FLAG_BUZZWOLE_HIDE_BACK_TO_BACK_TRAINER_2                   0x495 // Unused Flag
 #define FLAG_UNUSED_0x496                                           0x496 // Unused Flag
 #define FLAG_UNUSED_0x497                                           0x497 // Unused Flag
 #define FLAG_UNUSED_0x498                                           0x498 // Unused Flag
diff --git a/include/constants/opponents.h b/include/constants/opponents.h
index 9e7131077c..599f5447ce 100644
--- a/include/constants/opponents.h
+++ b/include/constants/opponents.h
@@ -858,12 +858,14 @@
 #define TRAINER_LEAF                        852
 #define TRAINER_BRENDAN_PLACEHOLDER         853
 #define TRAINER_MAY_PLACEHOLDER             854
+#define TRAINER_BUZZWOLE_GRUNT_1            855
+#define TRAINER_BUZZWOLE_GRUNT_2            856
 
-// NOTE: Because each Trainer uses a flag to determine when they are defeated, there is only space for 9 additional trainers before trainer flag space overflows
+// NOTE: Because each Trainer uses a flag to determine when they are defeated, there is only space for 9 (now 7 since scarlet just added 2) additional trainers before trainer flag space overflows
 //       More space can be made by shifting flags around in constants/flags.h or changing how trainer flags are handled
 //       MAX_TRAINERS_COUNT can be increased but will take up additional saveblock space
 
-#define TRAINERS_COUNT                      855
+#define TRAINERS_COUNT                      857
 #define MAX_TRAINERS_COUNT                  864
 #define TRAINER_PARTNER(partner)           (MAX_TRAINERS_COUNT + partner)
 
diff --git a/src/data/trainers.party b/src/data/trainers.party
index 9cd91085b2..1bcc8ddc2f 100644
--- a/src/data/trainers.party
+++ b/src/data/trainers.party
@@ -235,6 +235,42 @@ Poochyena
 Level: 9
 IVs: 0 HP / 0 Atk / 0 Def / 0 SpA / 0 SpD / 0 Spe
 
+=== TRAINER_BUZZWOLE_GRUNT_1 ===
+Name: GRUNT
+Class: Team Aqua
+Pic: Aqua Grunt M
+Gender: Male
+Music: Aqua
+Double Battle: No
+AI: Check Bad Move
+
+Carvanha
+Level: 15
+IVs: 24 HP / 24 Atk / 24 Def / 24 SpA / 24 SpD / 24 Spe
+- Bite
+- Leer
+- Aqua Jet
+- Water Gun
+
+
+=== TRAINER_BUZZWOLE_GRUNT_2 ===
+Name: GRUNT
+Class: Team Aqua
+Pic: Aqua Grunt F
+Gender: Female
+Music: Aqua
+Double Battle: No
+AI: Check Bad Move
+
+Poochyena
+Level: 15
+IVs: 24 HP / 24 Atk / 24 Def / 24 SpA / 24 SpD / 24 Spe
+- Bite
+- Howl
+- Tackle
+- Leer
+
+
 === TRAINER_MARCEL ===
 Name: MARCEL
 Class: Cooltrainer
diff --git a/src/data/wild_encounters.json b/src/data/wild_encounters.json
index 290b037c01..960e0a7519 100755
--- a/src/data/wild_encounters.json
+++ b/src/data/wild_encounters.json
@@ -56,8 +56,6 @@
             1
           ],
           "groups": {
-            "old_rod": [],
-            "good_rod": [],
             "super_rod": [
               0,
               1,
@@ -11562,88 +11560,88 @@
         {
           "map": "MAP_LITTLEROOT_TOWN",
           "base_label": "gLittlerootTown",
-          "fishing_mons": {
-            "encounter_rate": 30,
+          "water_mons": {
+            "encounter_rate": 4,
             "mons": [
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_SURSKIT"
+                "species": "SPECIES_SQUIRTLE"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_SURSKIT"
+                "species": "SPECIES_TOTODILE"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_LOTAD"
+                "species": "SPECIES_OSHAWOTT"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_LOTAD"
+                "species": "SPECIES_FROAKIE"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_GOLDEEN"
-              },
+                "species": "SPECIES_FROAKIE"
+              }
+            ]
+          },
+          "fishing_mons": {
+            "encounter_rate": 30,
+            "mons": [
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_GOLDEEN"
+                "species": "SPECIES_SURSKIT"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_POLIWAG"
+                "species": "SPECIES_SURSKIT"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_TYMPOLE"
+                "species": "SPECIES_LOTAD"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_TYMPOLE"
+                "species": "SPECIES_LOTAD"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_TYMPOLE"
-              }
-            ]
-          },
-          "water_mons": {
-            "encounter_rate": 4,
-            "mons": [
+                "species": "SPECIES_GOLDEEN"
+              },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_SQUIRTLE"
+                "species": "SPECIES_GOLDEEN"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_TOTODILE"
+                "species": "SPECIES_POLIWAG"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_OSHAWOTT"
+                "species": "SPECIES_TYMPOLE"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_FROAKIE"
+                "species": "SPECIES_TYMPOLE"
               },
               {
                 "min_level": 5,
                 "max_level": 5,
-                "species": "SPECIES_FROAKIE"
+                "species": "SPECIES_TYMPOLE"
               }
             ]
           }
-- 
2.52.0.windows.1

